commands/util.go

```diff
@@ -2,35 +2,82 @@ package commands
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/bndw/pick/backends"
 	"github.com/bndw/pick/crypto"
+	"github.com/bndw/pick/errors"
 	"github.com/bndw/pick/safe"
 	"github.com/bndw/pick/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
-func loadSafe() (*safe.Safe, error) {
-	password, err := utils.GetPasswordInput("Enter your master password")
-	if err != nil {
-		return nil, err
+func runCommand(c func([]string, *pflag.FlagSet) error, cmd *cobra.Command, args []string) {
+	if err := c(args, cmd.Flags()); err != nil {
+		if _, isUsageErr := err.(*errors.InvalidCommandUsage); isUsageErr {
+			cmd.Usage()
+			os.Exit(1)
+		}
+		os.Exit(handleError(err))
 	}
+	os.Exit(0)
+}
+
+type safeLoader struct {
+	password     *[]byte
+	maxLoadTries int
+	loadTries    int
+}
+
+func newSafeLoader() *safeLoader {
+	return &safeLoader{
+		maxLoadTries: 0,
+	}
+}
+
+func (sl *safeLoader) RememberPassword() {
+	sl.maxLoadTries++
```

Comment:
This one might not be obvious: Per https://github.com/bndw/pick/pull/109/files#diff-c07200a8f18f2a355ec0bd360f843b40R72 we ask for a new password if `sl.maxLoadTries > sl.loadTries`. `RememberPassword` increases `maxLoadTries` which allows to reuse the old password once more.